### DISCUSSION TITLE
Update document how to authenticate when user graphql

### DISCUSSION THREAD

**Discussion by tranlehaiquan** [keystonejs/keystone#7482] [Feature requests]
I can't see any guide on how to use `sessionToken` after use `authenticateUserWithPassword`. I see that we using header cookie with `keystonejs-session` and headers Authorization Bearer.

![Screen Shot 2022-04-24 at 16 35 12](https://user-images.githubusercontent.com/17347993/164970971-343053a6-1f28-4739-8337-0237bee179bf.png)

Could anybody make an update on the document? 
(I also see the in roadmap we have this, but don't know if this included or not)
![Screen Shot 2022-04-24 at 16 52 59](https://user-images.githubusercontent.com/17347993/164971009-6f8397b4-cf59-4160-9af8-3c4c7f4d783d.png)

**Comment by raveling** [comment]
@tranlehaiquan just acknowledging that this is a gap in our docs, and we've added it to our backlog.

**Reply by jschuur** [reply]
Just ran into a bit of a struggle with the docs on how to make an authenticated GraphQL request too and ultimately came up with this approach:

```js
import 'dotenv/config';

import { gql, GraphQLClient } from 'graphql-request';

const buildsQuery = gql`
  query {
    builds {
      id
      name
    }
  }
`;

const authMutation = gql`
  mutation {
    authenticateUserWithPassword(
    email: "${process.env.ADMIN_USER}"
    password: "${process.env.ADMIN_PASSWORD}"
    ) {
    ... on UserAuthenticationWithPasswordSuccess {
        sessionToken
    }
    }
  }
`;

console.log(`Using ${process.env.GRAPHQL_API_ENDPOINT}`);
const graphQLClient = new GraphQLClient(process.env.GRAPHQL_API_ENDPOINT);

(async () => {
  let res;

  res = await graphQLClient.request(authMutation);
  graphQLClient.setHeader('authorization',
    `Bearer ${res.authenticateUserWithPassword.sessionToken}`);

  res = await graphQLClient.request(buildsQuery);
  console.log(res);
})();
```

Something more explicit in the docs about putting the session token into the header for subsequent requests would definitely be useful.

**Reply by flexdinesh** [reply]
Thanks @jschuur. We're gonna update our docs to cover _how to auth with the default API_. We're building a bunch of new end-to-end examples to help with cases like these as well. 🤞 we'll get them out in a few weeks.
